### PR TITLE
compose: add upgrade notes for 2.17.x

### PR DESCRIPTION
**Pull request by milas**
### Proposed changes
Compose had several correctness changes in recent releases, which have caught folks off-guard. Added an advisory "Upgrade Notes" section to the v2.17.0 release section with the highlights.

I also removed a feature (`dockerfile_inline`) because there is a bug in the initial implementation rendering it unusable for the moment, so we will (re-)announce it once fixed.

### Related issues (optional)
* https://github.com/docker/compose/issues/10411
* https://github.com/docker/compose/issues/10407
* https://github.com/docker/compose/issues/10394
* https://github.com/docker/compose/issues/10405
* https://github.com/docker/compose/issues/10426
* https://github.com/docker/compose/issues/10431
* https://github.com/docker/compose/issues/10465

### CODE REVIEW

**Review by StefanScherer — APPROVED** [review]

File: compose/release-notes.md [hunk 1]
```diff
@@ -34,6 +34,11 @@ redirect_from:
 
 ## 2.17.0
 {% include release-date.html date="2023-03-23" %}
+### Upgrade Notes
```
Comment (aevesdocker) [inline compose/release-notes.md:37]:
```suggestion
### Upgrade notes
```

File: compose/release-notes.md [hunk 2]
```diff
@@ -34,6 +34,11 @@ redirect_from:
...
+- Project name validation is more strictly enforced. Project names can only include letters, numbers, `_`, `-` and must be lowercase and start with a letter or number
```
Comment (aevesdocker) [inline compose/release-notes.md:38]:
```suggestion
- Project name validation is more strictly enforced. Project names can only include letters, numbers, `_`, `-` and must be lowercase and start with a letter or number.
```

File: compose/release-notes.md [hunk 3]
```diff
@@ -34,6 +34,11 @@ redirect_from:
...
+- Boolean fields in YAML must be either `true` or `false`. Deprecated YAML 1.1 values such as "on" or "no" are not supported
```
Comment (aevesdocker) [inline compose/release-notes.md:39]:
```suggestion
- Boolean fields in YAML must be either `true` or `false`. Deprecated YAML 1.1 values such as "on" or "no" are not supported.
```

File: compose/release-notes.md [hunk 4]
```diff
@@ -34,6 +34,11 @@ redirect_from:
...
+- Duplicate YAML merge keys (`<<`) are rejected
```
Comment (aevesdocker) [inline compose/release-notes.md:40]:
```suggestion
- Duplicate YAML merge keys (`<<`) are rejected.
```

**Review by aevesdocker — APPROVED** [review]
Thank you @milas

**Pull request merged**